# HG changeset patch
# User Rik <rik@octave.org>
# Date 1416502301 28800
#      Thu Nov 20 08:51:41 2014 -0800
# Node ID 3a905b642408ea246861dc30dc8bc1be8867d530
# Parent  bb8d3f17248dd363f3d50412f3ef3f7e98b6785e
savepath.m: Fix failing BIST test

* savepath.m: Check that the path to save is not empty before
attempting to save it.

diff --git a/scripts/path/savepath.m b/scripts/path/savepath.m
--- a/scripts/path/savepath.m
+++ b/scripts/path/savepath.m
@@ -131,22 +131,27 @@ function retval = savepath (file)
     if (! isempty (default_path))
       n1 = find (strcmp (default_path{1}, path_to_preserve));
       n2 = find (strcmp (default_path{end}, path_to_preserve));
       n_middle = round ((n1+n2)/2);
       [~, n] = setdiff (path_to_preserve, default_path);
       path_to_save = path_to_preserve(sort (n));
       ## Remove pwd
       path_to_save(strcmp (path_to_save, ["." pathsep])) = [];
-      n = ones (size (path_to_save));
-      for m = 1:numel (path_to_save)
-        n(m) = find (strcmp (path_to_save{m}, path_to_preserve));
-      endfor
-      path_to_save_begin = path_to_save(n <= n_middle);
-      path_to_save_end   = path_to_save(n > n_middle);
+      if (! isempty (path_to_save))
+        n = ones (numel (path_to_save), 1);
+        for m = 1:numel (path_to_save)
+          n(m) = find (strcmp (path_to_save{m}, path_to_preserve));
+        endfor
+        path_to_save_begin = path_to_save(n <= n_middle);
+        path_to_save_end   = path_to_save(n > n_middle);
+      else
+        path_to_save_begin = {};
+        path_to_save_end   = {};
+      endif
     else
       path_to_save_begin = path_to_preserve;
       path_to_save_end   = {};
     endif
     path_to_save_begin = cell2mat (path_to_save_begin);
     path_to_save_end   = cell2mat (path_to_save_end);
 
     ## Use single quotes for PATH argument to avoid string escape
